src/extension.ts

```diff
@@ -63,3 +80,77 @@ const getComponentName = (): Thenable<string | undefined> =>
         : "Please enter a valid component name";
     },
   });
+
+/**
+ * Check availability of Angular CLI on the command line
+ * @param directory Directory which will use the Angular CLI
+ */
+const isAngularCliAvailable = (directory: string): boolean => {
+  try {
+    execSync("ng --version", { cwd: directory });
+    return true;
+  } catch {
+    return false;
+  }
+};
+
+/**
+ * Get the name of the directory of the current file
+ * @returns Name of the directory
+ */
+const getDirectoryName = (): string | undefined => {
```

Comment:
(Optional) We actually already have the `document`, so you could pass it into the function instead of reading it again

Reply:
Yep, that's better.